tests/StringViewTest.cpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/Api.h>
+#include <aws/crt/StringView.h>
+#include <aws/crt/Types.h>
+#include <aws/testing/aws_test_harness.h>
+
+static int s_test_string_view(struct aws_allocator *allocator, void *ctx)
```

Comment:
Should we a) break these into separate tests, and b) test conversion to/from std::string_view and friends?

Reply:
Yeah, you are write, we should support conversion from std::string_view

Reply:
When we do this, we should probably check for std::string_view support like this:
```
#if __cplusplus >= 201703L || _MSC_LANG >= 201703L
```

As explained on [this page](https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/) MSVC will only set the proper value for `__cplusplus` if the user adds the advanced `/Zc:__cplusplus` option to their program.